Assets/Scripts/PlayerSelect/PlayerSelectManager.cs

```diff
@@ -0,0 +1,159 @@
+
+    private PlayerInputManagerController playerInputManagerController;
+    private List<TMP_Text> playerNames;
+    private List<Animator> playerAnimators;
+    private int cardPeekCounter = 0;
+
+    public void Awake()
+    {
+        
+        playerNames = new List<TMP_Text>();
+        playerAnimators = new List<Animator>();
+    }
+
+    private void Start()
+    {
+        playerInputManagerController = PlayerInputManagerController.Singleton;
+        for (int i = 0; i < playerModels.Count; i++)
+        {
+            playerAnimators.Add(playerModels[i].GetComponentInChildren<Animator>()); // Add all animators to list
+        }
+    }
+
+    /// <summary>
+    /// Called when player is added. Sets the corresponding playermodel to active and fills in the playername in the nametag.
+    /// </summary>
+    /// <param name="playerName"></param>
+    /// <param name="color"></param>
+    /// <param name="playerID"></param>
+    public void setupPlayerSelectModels(string playerName, Color color, int playerID)
+    {
+        playerModels[playerID].GetComponentInChildren<SkinnedMeshRenderer>().material.color = color; // Set player model color
+        playerModels[playerID].SetActive(true); // Show corresponding player model
+        playerModels[playerID].transform.LookAt(new Vector3(playerSelectCam.transform.position.x, playerModels[playerID].transform.position.y, playerSelectCam.transform.position.z)); // Orient player model to look at camera
+        setPlayerNameTag(playerModels[playerID], playerName, playerID); // Create and display player nametag
+    }
+
+    /// <summary>
+    /// Fills in the playername and positions it.
+    /// </summary>
+    /// <param name="player"></param>
+    /// <param name="playerName"></param>
+    /// <param name="playerID"></param>
+    public void setPlayerNameTag(GameObject player, string playerName, int playerID)
+    {
+        // Check if nameTag exists already
+        bool nameExists = false;
+        for (int i = 0; i < playerNames.Count; i++)
+        {
+            if (playerNames[i].text == playerName)
+            {
+                nameExists = true;
+            }
+        }
+
+        // If it doesn't, create new nametag based on player name
+        if (!nameExists)
+        {
+            Vector3 playerPosition = new Vector3(player.transform.position.x, 2, player.transform.position.z);
+
+            TMP_Text name = nameTags[playerID];
+            name.GetComponent<Transform>().position = playerPosition; // Sets the nametag position over the head of the player model
+            name.text = playerName; // Sets the text of the nametag
+            playerNames.Add(name); // Adds TMP_text object to list for monitoring
+        }
+    }
+
+    /// <summary>
+    /// Starts or stops the playerselect animations coroutine. Called by "Start" and "Back" buttons in menu.
+    /// </summary>
+    /// <param name="play"></param>
+    public void controlAnimations(bool play)
```

Comment:
Why not have two methods to Play() and Pause() separately?

Reply:
Because i have a bad habit of concentrating code in my methods. Fixed for clarity. :)